lib/Tooling/libSwiftScan/SwiftCaching.cpp

```diff
@@ -920,141 +918,23 @@ static llvm::Error replayCompilation(SwiftScanReplayInstance &Instance,
-  const FrontendOptions &FrontendOpts =
-      Inst.getInvocation().getFrontendOptions();
-
-  swift::cas::CachedResultLoader Loader(CAS, Comp.Output);
-  if (auto Err = Loader.replay(
-          [&](file_types::ID Kind, llvm::cas::ObjectRef Ref) -> Error {
-            auto OutputPath = Outputs.find(Kind);
-            if (OutputPath == Outputs.end())
-              return createStringError(
-                  inconvertibleErrorCode(),
-                  "unexpected output kind in the cached output");
-            auto Proxy = CAS.getProxy(Ref);
-            if (!Proxy)
-              return Proxy.takeError();
-
-            if (Kind == file_types::ID::TY_CachedDiagnostics) {
-              assert(!DiagnosticsOutput && "more than 1 diagnostics found");
-              DiagnosticsOutput = std::move(*Proxy);
-            } else if (Kind == file_types::ID::TY_SymbolGraphFile &&
-                       !FrontendOpts.SymbolGraphOutputDir.empty()) {
-              auto Err = Proxy->forEachReference([&](llvm::cas::ObjectRef Ref)
-                                                     -> llvm::Error {
-                auto Proxy = CAS.getProxy(Ref);
-                if (!Proxy)
-                  return Proxy.takeError();
-                auto PathRef = Proxy->getReference(0);
-                auto ContentRef = Proxy->getReference(1);
-                auto Path = CAS.getProxy(PathRef);
-                auto Content = CAS.getProxy(ContentRef);
-                if (!Path)
-                  return Path.takeError();
-                if (!Content)
-                  return Content.takeError();
-
-                SmallString<128> OutputPath(FrontendOpts.SymbolGraphOutputDir);
-                llvm::sys::path::append(OutputPath, Path->getData());
-
-                OutputProxies.emplace_back(OutputEntry{
-                    std::string(OutputPath), std::move(*Content), Kind});
-
-                return Error::success();
-              });
-              if (Err)
-                return Err;
-            } else
-              OutputProxies.emplace_back(
-                  OutputEntry{OutputPath->second, std::move(*Proxy), Kind});
-            return Error::success();
-          }))
-    return Err;
-
-  // Replay diagnostics first.
-  // FIXME: Currently, the diagnostics is replay from the first file.
-  if (DiagnosticsOutput) {
-    DH.initDiagConsumers(Invocation);
-    DH.beginMessage(Invocation, Instance.Args);
-
-    if (auto E = CDP->replayCachedDiagnostics(DiagnosticsOutput->getData())) {
-      DH.endMessage(/*ReturnCode=*/1);
-      Inst.getDiags().finishProcessing();
-      return E;
-    }
-
-    if (Remarks)
-      Inst.getDiags().diagnose(SourceLoc(), diag::replay_output,
-                               "<cached-diagnostics>",
-                               CAS.getID(Comp.Key).toString());
-  } else {
-    // Don't write anything when parseable output is requested.
-    if (Invocation.getFrontendOptions().FrontendParseableOutput)
```

Comment:
I don't see this handled explicitly in the new code; does it happen some other way?

Reply:
I think that was not needed due to `DiagnosticHelper::Implementation::beginMessage()` already handling the case.

Reply:
That is kind of done in `beginMessage` already. I don't think this is needed.